### PR TITLE
update agnhost image to pull from registry.k8s.io

### PR DESCRIPTION
**Pull request by kwiesmueller**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Move the agnhost image in e2e tests from the staging repo to be pulled from `registry.k8s.io/e2e-test-images/agnhost:2.40` instead.

#### Which issue(s) this PR fixes:

Fixes a code TODO.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/assign @jbartosik

### CODE REVIEW

**Comment by kwiesmueller** [comment]
Running e2e tests locally right now.

**Review by jbartosik — APPROVED** [review]
/lgtm
/approve
/hold

Looks good to me but currently e2e tests are failing. I want to wait for rollback PR #5897 to merge and for a green e2e run before merging this (I don't want to risk having multiple sources of error, it makes debugging much harder).

**Comment by jbartosik** [comment]
/hold cancel

Tests are passing now

**Pull request merged**
